internal/helm-storage-backend/release_test.go

```diff
@@ -0,0 +1,497 @@
+package helmstoragebackend
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"helm.sh/helm/v3/pkg/time"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"capact.io/capact/internal/logger"
+	"capact.io/capact/internal/ptr"
+	pb "capact.io/capact/pkg/hub/api/grpc/storage_backend"
+)
+
+func TestRelease_GetValue_Success(t *testing.T) {
+	tests := []struct {
+		name string
+
+		givenDriver          *string
+		givenTypeInstanceID  *string
+		givenResourceVersion int
+		expectedDriver       string
+	}{
+		{
+			name:                "should use default driver and return the latest release",
+			givenTypeInstanceID: ptr.String("123"),
+			givenDriver:         nil,
+			expectedDriver:      "secrets",
+		},
+		{
+			name:                "should use configmap driver and return the latest release",
+			givenTypeInstanceID: ptr.String("123"),
+			givenDriver:         ptr.String("configmaps"),
+			expectedDriver:      "configmaps",
+		},
+		{
+			name:                 "should ignore resourceVersion and return the latest release",
+			givenTypeInstanceID:  ptr.String("123"),
+			givenResourceVersion: 42, // should be ignored
+			expectedDriver:       "secrets",
+		},
+		{
+			name:           "should return the latest release even if TypeInstance's id and revision are not specified",
```

Comment:
I'm not really sure if that's a useful test case - consider removing it. Our contract says that there should be TypeInstance ID, so I don't think we should even support this case. However, I'm not saying we should add an additional check for non-empty TypeInstance ID either, but I would simply remove this test case 😄